### PR TITLE
Small fixes related to BF16 fusion_gru and fusion_lstm

### PR DESCRIPTION
**Pull request by wozna**
### PR types

Bug fixes
### PR changes

OPs 
### Describe

 In this PR there are small fixes related to fusion_gru and fusion_lstm:
- added rewriting of the `use_mkldnn` attribute in `fc_gru_fuse_pass` and `fc_lstm_fuse_pass` and tests for that. Thanks to this, I could remove `mkldnn_placement_pass` in test  paddle/fluid/inference/tests/api/analyzer_lexical_analysis_gru_tester.cc  that runs twice. 
- added  `fusion_lstm`  to the list of bf16 operators
- added the `mkldnn_data_type` attribute to `fusion_lstm`, which is necessary to run the model on bfloat16

### CODE REVIEW

File: paddle/fluid/framework/ir/fc_gru_fuse_pass.cc [hunk 1]
```diff
@@ -48,7 +48,8 @@ static int BuildFusion(Graph* graph, const std::string& name_scope,
 
   // Create New OpDesc
   auto gru_creater = [&](Node* gru, Node* x, Node* weight_x, Node* weight_h,
-                         Node* bias, Node* hidden, Node* fc_bias) {
+                         Node* bias, Node* hidden, Node* fc_bias,
+                         const bool& use_mkldnn) {
```
Comment (jakpiase) [inline paddle/fluid/framework/ir/fc_gru_fuse_pass.cc]: Maybe passing use_mkldnn as value would be better? Same with lstm_creater

Comment (wozna) [reply paddle/fluid/framework/ir/fc_gru_fuse_pass.cc]: You are right, for such small types, it is better to pass by value.

**Comment by lidanqing-vv** [comment]
@wozna  Hi is this PR ready to review? If yes please ask members to review? because last time we informed Baidu that at least 2 LGTM the PR will be merged. Because I m not sure if this is ready so if it's ready you ask other members to review ok? Thanks

**Comment by lidanqing-vv** [comment]
LGTM. Thank you very much the fix! And thank you for calling and explaining it, the reviewing process get much faster.

@juncaipeng Could you please review and merge this PR. If you like, we can call and speed up the reviewing process.

**Review by juncaipeng — DISMISSED** [review]
LGTM

File: paddle/fluid/framework/ir/fc_gru_fuse_pass.cc [hunk 2]
```diff
@@ -48,7 +48,8 @@ static int BuildFusion(Graph* graph, const std::string& name_scope,
 
   // Create New OpDesc
   auto gru_creater = [&](Node* gru, Node* x, Node* weight_x, Node* weight_h,
```
Comment (jczaja) [inline paddle/fluid/framework/ir/fc_gru_fuse_pass.cc]: I found the name "gru_creater" a bit confusing. Maybe you could use "gru_creator" ?

Comment (wozna) [reply paddle/fluid/framework/ir/fc_gru_fuse_pass.cc]: Oh yes, it could be a spelling mistake.

**Review by jczaja — DISMISSED** [review]
LGTM

File: paddle/fluid/framework/ir/fc_gru_fuse_pass.cc [hunk 3]
```diff
@@ -149,21 +151,25 @@ static int BuildFusion(Graph* graph, const std::string& name_scope,
       LOG(INFO) << "fc_gru_fuse_pass not supported when origin_mode=True.";
       return;
     }
+    const bool use_mkldnn =
+        mul->Op()->GetAttrIfExists<bool>("use_mkldnn") &&
+        gru->Op()->GetAttrIfExists<std::string>("activation") == "tahn" &&
```
Comment (arlesniak) [inline paddle/fluid/framework/ir/fc_gru_fuse_pass.cc]: I think `tanh` should be here.

File: paddle/fluid/framework/ir/fc_lstm_fuse_pass.cc [hunk 4]
```diff
@@ -148,21 +149,29 @@ int BuildFusion(Graph* graph, const std::string& name_scope, Scope* scope,
     GET_IR_NODE_FROM_SUBGRAPH(Cell, Cell, lstm_pattern);
     GET_IR_NODE_FROM_SUBGRAPH(w, w, fc_pattern);
     GET_IR_NODE_FROM_SUBGRAPH(mul, mul, fc_pattern);
+    const bool use_mkldnn =
+        mul->Op()->GetAttrIfExists<bool>("use_mkldnn") &&
+        lstm->Op()->GetAttrIfExists<std::string>("gate_activation") ==
+            "sigmoid" &&
+        lstm->Op()->GetAttrIfExists<std::string>("cell_activation") == "tahn" &&
```
Comment (arlesniak) [inline paddle/fluid/framework/ir/fc_lstm_fuse_pass.cc]: Typo `tanh`

File: paddle/fluid/framework/ir/fc_lstm_fuse_pass.cc [hunk 5]
```diff
@@ -148,21 +149,29 @@ int BuildFusion(Graph* graph, const std::string& name_scope, Scope* scope,
...
+        lstm->Op()->GetAttrIfExists<std::string>("candidate_activation") ==
+            "tahn";
```
Comment (arlesniak) [inline paddle/fluid/framework/ir/fc_lstm_fuse_pass.cc]: Ditto

**Review by arlesniak — COMMENTED** [review]
Please double check the tests as they passed with typo.

File: paddle/fluid/framework/ir/fc_gru_fuse_pass.cc (hunk 3 above)
Comment (wozna) [reply paddle/fluid/framework/ir/fc_gru_fuse_pass.cc]: You're right, I'll check that

**Review by jczaja — APPROVED** [review]

**Comment by lidanqing-intel** [comment]
LGTM.

**Comment by lidanqing-intel** [comment]
@wozna Hi do you know why mul+elementwise_add is not fused as fc with this PR ?
image

**Review by lidanqing-vv — APPROVED** [review]
LGTM. Thanks a lot

**Pull request merged**